Microsoft.Toolkit.Uwp.UI.Animations/Expressions/ExpressionNodes/ExpressionNode.cs

```diff
@@ -0,0 +1,678 @@
+        /// Releases all resources used by this ExpressionNode.
+        /// </summary>
+        public void Dispose()
+        {
+            _objRefList = null;
+            _compObjToParamNameMap = null;
+            _constParamMap = null;
+            Subchannels = null;
+            PropertyName = null;
+            NodeType = ExpressionNodeType.Count;
+
+            // Note: we don't recursively dispose all child nodes, as those nodes could be in use by a different Expression
+            Children = null;
+
+            if (ExpressionAnimation != null)
+            {
+                ExpressionAnimation.Dispose();
+                ExpressionAnimation = null;
+            }
+        }
+
+        /// <summary>
+        /// Creates the expression node.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns>T.</returns>
+        /// <exception cref="System.Exception">unexpected type</exception>
+        internal static T CreateExpressionNode<T>()
+            where T : class
+        {
+            T newNode;
+
+            if (typeof(T) == typeof(BooleanNode))
+            {
+                newNode = new BooleanNode() as T;
+            }
+            else if (typeof(T) == typeof(ScalarNode))
+            {
+                newNode = new ScalarNode() as T;
+            }
+            else if (typeof(T) == typeof(Vector2Node))
+            {
+                newNode = new Vector2Node() as T;
+            }
+            else if (typeof(T) == typeof(Vector3Node))
+            {
+                newNode = new Vector3Node() as T;
+            }
+            else if (typeof(T) == typeof(Vector4Node))
+            {
+                newNode = new Vector4Node() as T;
+            }
+            else if (typeof(T) == typeof(ColorNode))
+            {
+                newNode = new Vector4Node() as T;
+            }
+            else if (typeof(T) == typeof(QuaternionNode))
+            {
+                newNode = new Vector4Node() as T;
+            }
+            else if (typeof(T) == typeof(Matrix3x2Node))
+            {
+                newNode = new Matrix4x4Node() as T;
+            }
+            else if (typeof(T) == typeof(Matrix4x4Node))
+            {
+                newNode = new Matrix4x4Node() as T;
+            }
+            else
+            {
+                throw new Exception("unexpected type");
+            }
+
+            return newNode;
+        }
+
+        /// <summary>
+        /// Creates the value keyword.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="keywordKind">Kind of the keyword.</param>
+        /// <returns>T.</returns>
+        /// <exception cref="System.Exception">Invalid ValueKeywordKind</exception>
+        internal static T CreateValueKeyword<T>(ValueKeywordKind keywordKind)
+            where T : class
+        {
+            T node = CreateExpressionNode<T>();
+
+            (node as ExpressionNode).ParamName = null;
```

Comment:
If for some reason the wrong type is passed in then `CreateExpressionNode` will throw an exception so impossible for it to be null